pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/RangeIndexCreator.java

```diff
@@ -321,6 +322,7 @@ public void seal()
     try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(_rangeIndexFile));
         DataOutputStream header = new DataOutputStream(bos);
         FileOutputStream fos = new FileOutputStream(_rangeIndexFile);
```

Comment:
Not introduced in this PR, but `FileOutputStream` is opened twice, is this intentional?

Reply:
@Jackie-Jiang  wouldn't something like this be more sensible?

```
    try (FileOutputStream fos = new FileOutputStream(_rangeIndexFile);
        BufferedOutputStream bos = new BufferedOutputStream(fos);
        DataOutputStream header = new DataOutputStream(bos);
        DataOutputStream dataOutputStream = new DataOutputStream(bos)) {
```

Reply:
its not clear to me if these `DataOutputStream` instances require their own `bos`

Reply:
Going over the code again, and seems we need to maintain 2 offsets